content/blog/eng/operators-overview.md

```diff
@@ -0,0 +1,140 @@
+---
+authors: ["AndreaRR18"]
+date: "2020-03-24"
+draft: true
+share: true
+categories: [English, Operators]
+title: "Operators overview"
+type: "post"
+languageCode: "en-EN"
+toc: true
+---
+
+# Operators overview
+Operators are useful constructors and they are present in most programming languages. They are fundamental for much operations and in this article we will make an overview to better understand their properties.
+This article is the first of two articles about operators. In this article we will do an overview of the main operators' properties and in the second we will discuss about some custom operators that we use in our production code.
+
+# Definition
+An operator is a special symbol or expression that is used to check, change or combine values. Most programming languages support their construct and we can considered them like a functions but with a different syntax and semantic.
+Usually we use operators for arithmetics operations (+, -), logic (&&, ||) and comparison (==, >), but we also consider other types of operators including assignment and variable access operators. 
+Parameters that we pass to the operators are called operands.
+
+There are three different types of operators:
+* **Unary**: this type of operator takes a single parameter placed before (prefix) or after (postfix) it.
+* **Binary**: this operator takes two parameters, it is palced between them. It’s also called infix operator.
+* **Ternary**: this particular type of operator works with three parameters, the first operand is a predicate and based on its result evaluate the first operand or the second.
+
+# User defined operators
+Some programming languages, like (Haskell, F#, Swift, ...), allow to define custom operators with special symbols, some other language allows to define literal operators (Kotlin, Pascal, ...), in both cases we have to define a function that will be executed when we use some operator.
+Some programming languages allow operator overloading, this operation is identified like ad hoc polymorphism. An overloaded declaration is a declaration that is declared with the same name in the same scope, except that both declarations have different arguments and different implementation.
```

Comment:
Make emphasize on 'ad hoc polymorphism'

Reply:
what do you think of this explanation?

"ad hoc polymorphism, it is a kind of polymorphism in which an operator can be applied to arguments of different types because a polymorphic operator can denote a different implementations depending on the type of arguments."